base_layer/wallet/src/util/encryption.rs

```diff
@@ -77,7 +78,7 @@ pub fn decrypt_bytes_integral_nonce(
 pub fn encrypt_bytes_integral_nonce(
     cipher: &XChaCha20Poly1305,
     domain: Vec<u8>,
-    plaintext: Vec<u8>,
+    plaintext: Hidden<Vec<u8>>,
```

Comment:
Another possibility is to pass a plain `Vec<u8>` as input and then zeroize `plaintext` at the end of the function.